core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -301,15 +282,31 @@ public void push(Repo<T> repo) throws StackOverflowException {
 
       String txpath = getTXPath(fateId);
       try {
-        String top = findTop(txpath);
-        if (top != null && Long.parseLong(top.split("_")[1]) > 100) {
+        AtomicBoolean validDequeSize = new AtomicBoolean(false);
+
+        zk.mutateExisting(txpath, currSerFateData -> {
+          FateData<T> fateData = new FateData<>(currSerFateData);
+          Preconditions.checkState(REQ_PUSH_STATUS.contains(fateData.status),
+              "Tried to push to the repo stack for %s when the transaction status is %s", fateId,
+              fateData.status);
+          var repoDeque = fateData.repoDeque;
+          validDequeSize.set(repoDeque.size() < MAX_REPOS);
```

Comment:
StackOverflowException is an internal accumulo exception so we can easily change it.  Can change it to extend RuntimeException instead of Exception.  The could throw it from around this point in the code and that would simplify the code because could remove atomic boolean.

Reply:
Good suggestion. Changed StackOverflowException and fixed it's use here/avoid atomic bool.